### PR TITLE
Add basic wordpiece processor

### PR DESCRIPTION
**Pull request by danieldk**

### CODE REVIEW

File: cutlery/_wordpiece.pyx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            token = token[match.prefix_len:]
+            pieces = &self.continuation_pieces
+            prefix = "##"
+        return token_ids, token_pieces
+
+    def get_initial(self, piece: str) -> int:
+        cdef unordered_map[string, size_t].const_iterator iter
+        iter = self.initial_pieces.const_find(piece.encode('utf8'))
+        if iter == self.initial_pieces.end():
+            raise KeyError()
```
Comment (shadeMe) [inline cutlery/_wordpiece.pyx:48]: We could use a more descriptive error message here.

File: cutlery/_wordpiece.pyx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        else:
+            return deref(iter).second
+
+
+    @staticmethod
+    def from_file(str filename):
```
Comment (shadeMe) [inline cutlery/_wordpiece.pyx:54]: We should keep the type hints consistent across all `def` and `cdef` functions, I think. So, regular Python type hints for the former and C-style types for the latter?

File: cutlery/_wordpiece.pyx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        for piece, idx in self.initial_pieces:
+            pieces[idx] = piece.decode('utf8')
+
+        for piece, idx in self.continuation_pieces:
+            pieces[idx] = f"##{piece.decode('utf8')}"
+
+        return pieces
+
+
+cdef PieceMatch _find_longest_prefix(const unordered_map[string, size_t]* pieces, token: str):
```
Comment (shadeMe) [inline cutlery/_wordpiece.pyx:71]: Type hint like above.

File: cutlery/_wordpiece.pyx (hunk 1 above)
Comment (danieldk) [reply cutlery/_wordpiece.pyx]: Fixed.

File: cutlery/_wordpiece.pyx (hunk 3 above)
Comment (danieldk) [reply cutlery/_wordpiece.pyx]: Fixed.

File: cutlery/_wordpiece.pyx (hunk 2 above)
Comment (danieldk) [reply cutlery/_wordpiece.pyx]: Fixed.

**Review by shadeMe — APPROVED** [review]

**Pull request merged**